parsimony/functions/penalties.py

```diff
@@ -1251,13 +1260,19 @@ def lambda_max(self):
         # should allow dense matrices as well.
         if self._lambda_max is None:
 
-            from parsimony.algorithms.nipals import RankOneSparseSVD
-
-            A = self.A
-            # TODO: Add max_iter here!
-            v = RankOneSparseSVD().run(A)  # , max_iter=max_iter)
-            us = A.dot(v)
-            self._lambda_max = np.sum(us ** 2.0)
+            if self.A != None:
+                from parsimony.algorithms.nipals import RankOneSparseSVD
+                A = self.A
+                # TODO: Add max_iter here!
+                v = RankOneSparseSVD().run(A)  # , max_iter=max_iter)
+                us = A.dot(v)
+                self._lambda_max = np.sum(us ** 2.0)
+            else:
+                from scipy.sparse.linalg import svds
+                L = self.La.asfptype()
+                # TODO: is using RankOneSparseSVD more efficient here?
+                u,s,v = svds(L, k=1)
+                self._lambda_max = s[0]
```

Comment:
> Can you double check this?

Here it is not squared

Reply:
I think it should be squared here.. The method is supposed to return the largest eigenvalue, not a singular value.

Reply:
Exactly! So i squared here instead of at l1250 because of the case A given and l1269.